build.js

```diff
@@ -258,93 +266,118 @@ function getNumberOfParentFolders(path, root = "html") {
-			str_builder += card.getCode();
-			counter++;
-			if(counter === 3) {
+		// Checks performed
+		if(languages.includes('en'))
+			var object_en = JSON.parse(fs.readFileSync(enJson));
+		if(languages.includes('sk'))
+			var object_sk = JSON.parse(fs.readFileSync(skJson));
+
+		// Compiling interview modal navigation
+		fs.writeFileSync(getDirname() + "/components/" + "interview_navigation.html", transpileInterviewNavigation(object_en)); 
+
+		// Pre-compiling the English version of the interview cards
+		if(object_en) {
+			var str_builder = ``;
+			var counter = 0;
+			object_en.map(obj => {
+				// I check how many cards are already in one deck if it is the start of a new deck I manually add a card deck starting block
+				if(counter === 0)
+					str_builder += card_deck_start;
+				// Creating new instance of Interview Card so I can access it's HTML code
+				var card = new InterviewCard(obj.Image, obj.Title, obj.ShortInfo, obj.LongInfo, obj.PhotosFolderPath, obj.Name);
+				str_builder += card.getCode();
+				counter++;
+				// Here I state how many cards should be in one card deck or you can call it card row
+				if(counter === 3) {
+					str_builder += card_deck_end;
+					counter = 0;
+				}
+			});
+			// Here I check if the last card added to row was 3rd in the row if not I have to manually add card deck end block
+			if(counter !== 0) {
 				str_builder += card_deck_end;
 				counter = 0;
 			}
-		});
-		if(counter !== 0) {
-			str_builder += card_deck_end;
-			counter = 0;
+			var file = files.find(obj => {
+				return obj.language === 'en';
+			});
+			// Writing the final compiled HTML code into file
+			fs.writeFileSync(file.src, str_builder);
 		}
-		var file = files.find(obj => {
-			return obj.language === 'en';
-		});
-		fs.writeFileSync(file.src, str_builder);
-	}
-	if(obj_sk) {
-		var str_builder = ``;
-		var counter = 0;
-		obj_en.map(obj => {
-			if(counter === 0)
-				str_builder += card_deck_start;
-			var card = new InterviewCard(obj.Image, obj.Title, obj.ShortInfo, obj.LongInfo, obj.PhotosFolderPath, obj.Name);
-			str_builder += card.getCode();
-			counter++;
-			if(counter === 3) {
+
+		// Pre-compiling the Slovak version of the interview cards
+		if(object_sk) {
+			var str_builder = ``;
+			var counter = 0;
+			object_sk.map(obj => {
+				if(counter === 0)
+					str_builder += card_deck_start;
+				var card = new InterviewCard(obj.Image, obj.Title, obj.ShortInfo, obj.LongInfo, obj.PhotosFolderPath, obj.Name);
+				str_builder += card.getCode();
+				counter++;
+				if(counter === 3) {
+					str_builder += card_deck_end;
+					counter = 0;
+				}
+			});
+			if(counter !== 0) {
 				str_builder += card_deck_end;
 				counter = 0;
 			}
-		});
-		if(counter !== 0) {
-			str_builder += card_deck_end;
-			counter = 0;
+			var file = files.find(obj => {
+				return obj.language === 'sk';
+			});
+			fs.writeFileSync(file.src, str_builder);
```

Comment:
it is often a good pattern to separate code as much as possible that does processing without side effects and then all the code which has side effects (writing to files, etc.).
This means that one function would compute the HTML, the other would then write the needed files.